### PR TITLE
csv DictReader should only yield OrderedDicts in Py3.6

### PR DESCRIPTION
**Pull request by graingert**

### CODE REVIEW

**Comment by graingert** [comment]
although of course this is still wrong

out of the box DictReader supports returning floats and strings
restkey should be a generic type var of any hashable
restvar should be a generic type of any var

File: stdlib/2and3/csv.pyi [hunk 1]
```diff
@@ -52,34 +52,28 @@ if sys.version_info >= (3,):
...
+    restval = ...  # type: Optional[str]
+    reader = ...  # type: _reader
+    dialect = ...  # type: _Dialect
+    line_num = ...  # type: int
+    fieldnames = ...  # type: Sequence[str]
+    def __init__(self, f: Iterable[str], fieldnames: Sequence[str] = ...,
+                 restkey: Optional[str] = ..., restval: Optional[str] = ..., dialect: _Dialect = ...,
+                 *args: Any, **kwds: Any) -> None: ...
+    def __iter__(self) -> 'DictReader': ...
+    def __next__(self) -> _DRMapping: ...
```
Comment (JelleZijlstra) [inline stdlib/2and3/csv.pyi]: This is redundant, you have `__next__` below.

Comment (graingert) [inline stdlib/2and3/csv.pyi]: oh woops

**Comment by graingert** [comment]
@JelleZijlstra  ok, done

**Pull request merged**